### PR TITLE
Increment version number to v0.17.0

### PR DESCRIPTION
**Pull request by thisac**
Increments the version to v0.17.0.

Also, updates the changelog (adds some new entries) as well as the docstring for the `sf.about()` method.

### CODE REVIEW

File: strawberryfields/__init__.py [hunk 1]
```diff
@@ -79,17 +79,17 @@ def about():
...
+        Numpy version:             1.19.5
+        Scipy version:             1.4.1
+        SymPy version:             1.5.1
         NetworkX version:          2.4
-        The Walrus version:        0.10.0
-        Blackbird version:         0.2.1
-        TensorFlow version:        2.0.0
+        The Walrus version:        0.15.0-dev
+        Blackbird version:         0.3.1-dev
+        TensorFlow version:        2.4.0
```
Comment (thisac) [inline strawberryfields/__init__.py:92]: Noticed that these were a bit out-of-date. :smile:

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Release 0.17.0 (development release)
+# Release 0.17.0 (current release)
```
Comment (josh146) [inline .github/CHANGELOG.md:1]: 🥳

File: strawberryfields/__init__.py [hunk 3]
```diff
@@ -79,17 +79,17 @@ def about():
         Strawberry Fields: a Python library for continuous-variable quantum circuits.
         Copyright 2018-2020 Xanadu Quantum Technologies Inc.
 
-        Python version:            3.6.8
-        Platform info:             Linux-5.0.0-36-generic-x86_64-with-debian-buster-sid
-        Installation path:         /home/josh/Dropbox/Work/Xanadu/sf_cloud/strawberryfields
```
Comment (josh146) [inline strawberryfields/__init__.py:84]: Oops 😆

**Review by josh146 — APPROVED** [review]
Thanks for getting to this @thisac. This is actually quite a big release!

**Comment by antalszava** [comment]
[ch4266]

**Pull request merged**